nugen/NuReweight/GENIEReweight.cxx

```diff
@@ -345,55 +173,157 @@ namespace rwgt {
+          configure_NCEL_calc = true;
           break;
 
-        //CC QE Axial parameters
+        // CC QE Axial parameters - shape+rate mode
         case rwgt::fReweightNormCCQE:
-        case rwgt::fReweightNormCCQEenu:
         case rwgt::fReweightMaCCQEshape:
+        case rwgt::fReweightE0CCQEshape:
+          fMaCCQEModes |= CCQEKnobMode::RateShape;
+          configure_CCQE_calc = true;
+          break;
+
+        // CC QE Axial parameters - parameter mode
         case rwgt::fReweightMaCCQE:
-          fReweightQEMA = true;
+        case rwgt::fReweightE0CCQE:
+          fMaCCQEModes |= CCQEKnobMode::Parameter;
+          configure_CCQE_calc = true;
+          break;
+
+        // Z-expansion (axial) parameters
+        case rwgt::fReweightZNormCCQE:
+        case rwgt::fReweightZExpA1CCQE:
+        case rwgt::fReweightZExpA2CCQE:
+        case rwgt::fReweightZExpA3CCQE:
+        case rwgt::fReweightZExpA4CCQE:
+          fMaCCQEModes |= CCQEKnobMode::ZExp;
+          configure_CCQE_calc = true;
+          break;
+
+        // other CCQE effects
+        case rwgt::fReweightAxFFCCQEshape:   // note: doesn't go in "shape+rate" collection above; this knob changes the underlying shape in a non-rate-conserving way
+        case rwgt::fReweightRPA_CCQE:
+        case rwgt::fReweightCoulombCCQE:
+          configure_CCQE_calc = true;
           break;
 
-        //CC QE Vector parameters
+        // CC QE Vector parameters (dipole)
         case rwgt::fReweightVecCCQEshape:
-          fReweightQEVec = true;
+          configure_QEVec_calc = true;
           break;
 
-        //CC Resonance parameters
+        // Z-expansion (vector) parameters
+        case rwgt::fReweightZExpELFF:
+        case rwgt::fReweightZExpELFF_AP1:
+        case rwgt::fReweightZExpELFF_AP2:
+        case rwgt::fReweightZExpELFF_AP3:
+        case rwgt::fReweightZExpELFF_AP4:
+        case rwgt::fReweightZExpELFF_AN1:
+        case rwgt::fReweightZExpELFF_AN2:
+        case rwgt::fReweightZExpELFF_AN3:
+        case rwgt::fReweightZExpELFF_AN4:
+        case rwgt::fReweightZExpELFF_BP1:
+        case rwgt::fReweightZExpELFF_BP2:
+        case rwgt::fReweightZExpELFF_BP3:
+        case rwgt::fReweightZExpELFF_BP4:
+        case rwgt::fReweightZExpELFF_BN1:
+        case rwgt::fReweightZExpELFF_BN2:
+        case rwgt::fReweightZExpELFF_BN3:
+        case rwgt::fReweightZExpELFF_BN4:
+          configure_ZexpVector_calc = true;
+          break;
+
+        // MEC parameters
+        case rwgt::fReweightEmpMEC_Mq2d:
+        case rwgt::fReweightEmpMEC_Mass:
+        case rwgt::fReweightEmpMEC_Width:
+        case rwgt::fReweightEmpMEC_FracPN_NC:
+        case rwgt::fReweightEmpMEC_FracPN_CC:
+        case rwgt::fReweightEmpMEC_FracCCQE:
+        case rwgt::fReweightEmpMEC_FracNCQE:
+        case rwgt::fReweightEmpMEC_FracPN_EM:
+        case rwgt::fReweightEmpMEC_FracEMQE:
+          fMECTypes |= MECType::Empirical;
+          configure_MEC_calc = true;
```

Comment:
@chenel  I think you need a "break" here so as not to fall through to the next set of parameters, or is this intended to be `[[fallthrough]]`

Reply:
you are right!  good catch!!

Reply:
fixed in c623eaa